src/Stats.h

```diff
@@ -0,0 +1,23 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+namespace reactivesocket {
+// forward decl for NoopStats
+class NoopStats;
+
+class Stats {
+ public:
+  static NoopStats noop;
```

Comment:
Foo foo() {
  static Foo bar;
  return bar;
}

The difference is that bar is constructed at run time, the first time any code path traverses the line with `static ...`. Otherwise we risk https://isocpp.org/wiki/faq/ctors#static-init-order.